src/App.js

```diff
@@ -0,0 +1,250 @@
+      // let response = await fetch(url);
+      // let result = await response.json();
+      // console.log(result);
+      let response2 = await fetch(forecast);
+      let result2 = await response2.json();
+      console.log(result2);
+
+      this.setState(
+        {
+          location: city,
+          isLoading: false,
+          todaysDate: this.moment(result2.list[0].dt_txt).format("MM-DD-YYYY"),
+          foreCastArray: [
+            result2.list[0],
+            result2.list[8],
+            result2.list[16],
+            result2.list[24],
+            result2.list[32]
+          ]
+        },
+        () => {
+          console.log(this.state.foreCastArray);
+        }
+      );
+    } catch (e) {
+      console.log(e);
+    }
+  };
+  getLocation = (city = "Ho Chi Minh") => {
+    //default is hochiminh if no input specified
+    navigator.geolocation.getCurrentPosition(position => {
+      this.currentWeather(
+        city,
+        position.coords.latitude,
+        position.coords.longitude
+      ); //as of now, lat and long are not needed at all. Only if you use the api call which uses it
+    });
+  };
+  componentDidMount() {
+    //after component mounts and after 1st render
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        function(position) {
+          console.log("navigation.geolocation works");
+        },
+        function(error) {
+          console.log("Error occurred. Error code: " + error.code);
+        },
+        { timeout: 5000 }
+      );
+    } else {
+      console.log("no geolocation support");
+    }
+    this.getLocation();
+  }
+  changeCity = city => {
+    console.log("test");
+    this.getLocation(city);
+  };
+  render() {
+    if (this.state.isLoading === true) {
+      return (
+        <div className="text-center p-4">
+          <h1 className="text-light">I am Loading</h1>
+          <ClipLoader
+            css={
+              // Can be a string as well. Need to ensure each key-value pair ends with ;
+              `
+                display: block;
+                margin: 0 auto;
+                border-color: white;
+              `
+            }
+            size={150}
+            color={"#123abc"}
+            loading={this.state.isLoading}
+          />
+        </div>
+      );
+    } else
+      return (
+        <div className="p-5">
+          <div className="d-flex">
+            <div className="shadow text-center bg-light p-3 rounded w-50 my-3">
+              <h1 className="col-12 display-4 my-2 py-3">Weather App</h1>
+              <h2 className="col-12">
+                {this.state.location !== null && this.state.location}
+              </h2>
+              <h2 className="col-12">Date: {this.state.todaysDate}</h2>
+            </div>
+            <div className="d-inline-block flex-grow-1 p-3">
+              <div className="d-flex row justify-content-center align-items-center">
```

Comment:
Oof, a lot of repeated code here. A bit simpler would be something like:

at top of file 
```
const CITIES = ['London', "Paris', ... ]
```
then here

```
CITIES.map( city => 
   <button ....>
)
```